db/migrate/20210224190247_migrate_shipping_taxes.rb

```diff
@@ -56,13 +56,15 @@ def assign_to_shipping_methods
   end
   
   def migrate_tax_amounts_to_adjustments
+    shipping_tax_rates = Spree::TaxRate.where(tax_category: shipping_tax_category).to_a
+
     # Migrate all shipping tax amounts from shipment field to tax adjustments
     Spree::Adjustment.shipping.where("included_tax <> 0").includes(:source, :order).find_each do |shipping_fee|
       shipment = shipping_fee.source
       order = shipping_fee.order
       next if order.nil?
 
-      tax_rate = Spree::TaxRate.find_by(tax_category: shipping_tax_category, zone: order.tax_zone)
+      tax_rate = shipping_tax_rates.detect{ |rate| rate.zone == order.tax_zone }
```

Comment:
my point with the hash was that finding the rate will then become an operation with a constant cost whereas this needs to iterate them every time but that's probably negligible given the number of rates in that array :+1: